samples/net/sockets/big_http_download/src/big_http_download.c

```diff
@@ -119,9 +137,23 @@ void download(struct addrinfo *ai)
 
 	cur_bytes = 0;
 
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
+	sock = socket(ai->ai_family, ai->ai_socktype, IPPROTO_TLS_1_2);
+#else
 	sock = socket(ai->ai_family, ai->ai_socktype, ai->ai_protocol);
+#endif
 	CHECK(sock);
 	printf("sock = %d\n", sock);
+
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
+	sec_tag_t sec_tag_list[] = {
```

Comment:
I just got myself confused. So, we have `typedef sec_tag_t`, we have `struct sec_tag_list`. And here we have `sec_tag_t sec_tag_list`. I'd suggest to clean up naming of what's `sec_tag_list` list is.

Reply:
Oh, we also have NET_OPT_TLS_SEC_TAG_LIST, which doesn't really take `struct sec_tag_list`.